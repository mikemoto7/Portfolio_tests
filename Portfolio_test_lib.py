
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







