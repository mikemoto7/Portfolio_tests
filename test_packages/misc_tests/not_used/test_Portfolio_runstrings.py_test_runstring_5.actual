
Processing: user_runstring = ./Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml

Processing: --price_web_update

Processing:  --pt lp

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------
                    Sorted                                               Sorted                                               Sorted
Asset                Tot_%  Ret_%  Non_%     Asset                Tot_%   Ret_%  Non_%     Asset                Tot_%  Ret_%   Non_%
Small_Value           2.24   0.55   4.80     Mid_Blend_Aggress     8.55    0.00  21.52     Global                6.79   9.47    2.73
Small_Blend           2.66   0.85   5.42     Small_Value           2.24    0.55   4.80     Large_Value           5.56   6.38    4.31
Healthcare            3.04   0.81   6.43     Healthcare            3.04    0.81   6.43     Small_Growth          9.67  13.08    4.50
Large_Blend           5.13   4.53   6.02     Small_Blend           2.66    0.85   5.42     Small_Value           2.24   0.55    4.80
Large_Value           5.56   6.38   4.31     Large_Blend           5.13    4.53   6.02     Mid_Growth           10.36  13.91    4.97
Mid_Value             6.28   5.89   6.87     Mid_Value             6.28    5.89   6.87     Small_Blend           2.66   0.85    5.42
Global                6.79   9.47   2.73     Large_Value           5.56    6.38   4.31     Energy               10.68  14.11    5.46
Mid_Blend_Aggress     8.55   0.00  21.52     Global                6.79    9.47   2.73     Large_Blend           5.13   4.53    6.02
Small_Growth          9.67  13.08   4.50     Small_Growth          9.67   13.08   4.50     Healthcare            3.04   0.81    6.43
Mid_Growth           10.36  13.91   4.97     Mid_Growth           10.36   13.91   4.97     Mid_Value             6.28   5.89    6.87
Energy               10.68  14.11   5.46     Energy               10.68   14.11   5.46     Large_Growth         13.21  15.85    9.21
REIT                 12.57  14.56   9.53     REIT                 12.57   14.56   9.53     REIT                 12.57  14.56    9.53
Large_Growth         13.21  15.85   9.21     Large_Growth         13.21   15.85   9.21     Mid_Blend_Aggress     8.55   0.00   21.52
Individual_Stocks     3.27   0.00   8.22     Individual_Stocks     3.27    0.00   8.22     Individual_Stocks     3.27   0.00    8.22
Totals              100.01  99.99  99.99     Totals              100.01   99.99  99.99     Totals              100.01  99.99   99.99

