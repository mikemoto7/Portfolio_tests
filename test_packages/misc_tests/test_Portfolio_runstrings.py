import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






