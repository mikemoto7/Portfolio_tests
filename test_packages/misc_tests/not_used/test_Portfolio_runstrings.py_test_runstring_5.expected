
Processing: user_runstring = ./Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml

Processing: --price_web_update

Processing:  --pt lp

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------
                    Sorted                                               Sorted                                               Sorted
Asset                Tot_%  Ret_%   Non_%     Asset               Tot_%   Ret_%   Non_%     Asset               Tot_%  Ret_%   Non_%
Small_Value           2.24   0.55    4.80     Mid_Blend_Aggress    8.56    0.00   21.53     Global               6.76   9.43    2.73
Small_Blend           2.66   0.84    5.41     Small_Value          2.24    0.55    4.80     Large_Value          5.55   6.38    4.30
Healthcare            3.04   0.81    6.42     Healthcare           3.04    0.81    6.42     Small_Growth         9.67  13.09    4.49
Large_Blend           5.13   4.55    6.02     Small_Blend          2.66    0.84    5.41     Small_Value          2.24   0.55    4.80
Large_Value           5.55   6.38    4.30     Large_Blend          5.13    4.55    6.02     Mid_Growth          10.33  13.88    4.95
Mid_Value             6.28   5.88    6.88     Mid_Value            6.28    5.88    6.88     Small_Blend          2.66   0.84    5.41
Global                6.76   9.43    2.73     Large_Value          5.55    6.38    4.30     Energy              10.69  14.14    5.46
Mid_Blend_Aggress     8.56   0.00   21.53     Global               6.76    9.43    2.73     Large_Blend          5.13   4.55    6.02
Small_Growth          9.67  13.09    4.49     Small_Growth         9.67   13.09    4.49     Healthcare           3.04   0.81    6.42
Mid_Growth           10.33  13.88    4.95     Mid_Growth          10.33   13.88    4.95     Mid_Value            6.28   5.88    6.88
Energy               10.69  14.14    5.46     Energy              10.69   14.14    5.46     Large_Growth        13.14  15.76    9.17
REIT                 12.66  14.68    9.59     REIT                12.66   14.68    9.59     REIT                12.66  14.68    9.59
Large_Growth         13.14  15.76    9.17     Large_Growth        13.14   15.76    9.17     Mid_Blend_Aggress    8.56   0.00   21.53
Individual_Stocks     3.28   0.00    8.25     Individual_Stocks    3.28    0.00    8.25     Individual_Stocks    3.28   0.00    8.25
Totals               99.99  99.99  100.00     Totals              99.99   99.99  100.00     Totals              99.99  99.99  100.00

